### ISSUE TITLE
Enhancement: [no-misused-promises] Warn when spreading a promise

### ISSUE THREAD

**Issue by papermana** [typescript-eslint/typescript-eslint#5049]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-misused-promises

### Description

Please see [this issue](https://github.com/microsoft/TypeScript/issues/49161) in the TS repo.

### Fail

```typescript
console.log({ ...Promise.resolve({ key: 42 }) });
```

### Pass

```typescript
console.log({ ...await Promise.resolve({ key: 42 }) });
```

### Additional Info

I think I can implement this feature if it's wanted.
